web/src/main/java/org/openmrs/web/filter/update/UpdateFilter.java

```diff
@@ -334,54 +334,116 @@ protected boolean authenticateAsSuperUser(String usernameOrSystemId, String pass
 			connection = DatabaseUpdater.getConnection();
 			
 			String select = "select user_id, password, salt from users where (username = ? or system_id = ?) and retired = '0'";
-			PreparedStatement statement = connection.prepareStatement(select);
-			statement.setString(1, usernameOrSystemId);
-			statement.setString(2, usernameOrSystemId);
-			
-			if (statement.execute()) {
-				ResultSet results = statement.getResultSet();
-				if (results.next()) {
-					Integer userId = results.getInt(1);
-					DatabaseUpdater.setAuthenticatedUserId(userId);
-					String storedPassword = results.getString(2);
-					String salt = results.getString(3);
-					String passwordToHash = password + salt;
-					boolean result = Security.hashMatches(storedPassword, passwordToHash) && isSuperUser(connection, userId);
-					statement.close();
-					return result;
+			PreparedStatement statement = null;
+			try {
+				statement = connection.prepareStatement(select);
+				statement.setString(1, usernameOrSystemId);
+				statement.setString(2, usernameOrSystemId);
+
+				if (statement.execute()) {
+					ResultSet results = null;
+					try {
+						results = statement.getResultSet();
+						if (results.next()) {
+							Integer userId = results.getInt(1);
+							DatabaseUpdater.setAuthenticatedUserId(userId);
+							String storedPassword = results.getString(2);
+							String salt = results.getString(3);
+							String passwordToHash = password + salt;
+							boolean result = Security.hashMatches(storedPassword, passwordToHash)
+							        && isSuperUser(connection, userId);
+							connection.close();
+							results.close();
+							statement.close();
+							return result;
+						} else {
+							results.close();
+						}
+					}
+					catch (Exception resultsEx) {
+						if (results != null) {
+							try {
+								results.close();
+							}
+							catch (Exception resultsCloseEx) {
+								log.error("Failed to quietly close ResultSet", resultsCloseEx);
```

Comment:
do we only need to catch the exception and log? THere is no need of re throwing it?